src/util/CMakeLists.txt

```diff
@@ -33,12 +33,17 @@ file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/version.cpp.in
   "const char *CBMC_VERSION=\"@CBMC_RELEASE@ (@GIT_INFO@)\";\n")
 add_custom_target(
   generate_version_cpp
-  BYPRODUCTS version.cpp
+  BYPRODUCTS ${CBMC_SOURCE_DIR}/util/version.cpp
   COMMAND ${CMAKE_COMMAND}
     -D CBMC_SOURCE_DIR=${CBMC_SOURCE_DIR}
     -D CUR=${CMAKE_CURRENT_BINARY_DIR}
     -P ${CMAKE_BINARY_DIR}/version.cmake
-)
+    )
+
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND "${CMAKE_BUILD_TYPE}" STREQUAL "Release")
+  set_source_files_properties(format_number_range.cpp
+    PROPERTIES COMPILE_FLAGS "${CMAKE_CXX_FLAGS} -O2")
+endif()
```

Comment:
Perhaps we should instead rewrite the code in `format_number_range.cpp`? It could probably be replaced by multiple functions that handle the case of having/not having a value. Having a need for such excessive handling of possible error cases doesn't seem to be a good idea in a `static` function.

Reply:
I didn't really get rid of the error handling, but still did the refactoring in #6017 to make this warning go away.

Reply:
Let's see of the smoke comes out green with your version and O3 on Ubuntu 18